Tools/clinic/clinic.py

```diff
@@ -1250,18 +1250,32 @@ def parser_body(
             parser_prototype = self.PARSER_PROTOTYPE_VARARGS
             parser_definition = parser_body(parser_prototype, '    {option_group_parsing}')
 
-        elif not requires_defining_class and pos_only == len(parameters) - pseudo_args and clinic.limited_capi:
-            # positional-only for the limited C API
-            flags = "METH_VARARGS"
+        elif clinic.limited_capi:
+            if not requires_defining_class and pos_only == len(parameters) - pseudo_args:
+                # positional-only for the limited C API:
+                # PyArg_ParseTuple()
+                flags = "METH_VARARGS"
 
-            parser_prototype = self.PARSER_PROTOTYPE_VARARGS
-            parser_code = [normalize_snippet("""
-                if (!PyArg_ParseTuple(args, "{format_units}:{name}",
-                    {parse_arguments}))
-                    goto exit;
-            """, indent=4)]
-            argname_fmt = 'args[%d]'
-            declarations = ""
+                parser_prototype = self.PARSER_PROTOTYPE_VARARGS
+                parser_code = [normalize_snippet("""
+                    if (!PyArg_ParseTuple(args, "{format_units}:{name}",
+                        {parse_arguments}))
+                        goto exit;
+                """, indent=4)]
+                declarations = ""
+
+            else:
+                # positional-or-keyword arguments for the limited C API:
+                # PyArg_ParseTupleAndKeywords()
+                flags = "METH_VARARGS|METH_KEYWORDS"
+
+                parser_prototype = self.PARSER_PROTOTYPE_KEYWORD
+                parser_code = [normalize_snippet("""
+                    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "{format_units}:{name}", _keywords,
+                        {parse_arguments}))
+                        goto exit;
+                """, indent=4)]
+                declarations = "char* _keywords[] = {{{keywords_c} NULL}};"
```

Comment:
```suggestion
                declarations = "static char *_keywords[] = {{{keywords_c} NULL}};"
```

Reply:
I don't reallly understand the difference between the two. The value is constant so the variable should not really exist on the stack, no?